### PR TITLE
Introduce partial impl. of distributed repo

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: repository/src/raw.rs [hunk 1]
```diff
@@ -86,25 +95,20 @@ pub trait RawRepository: Send + Sync {
...
-    /// Create a commit with the given list to overwrite files.
-    ///
-    /// * `files`: the list of `(file_path, file_content_to_overwrite)`.
-    async fn create_commit_overwrite(
-        &mut self,
-        branch: &Branch,
-        commit_message: &str,
-        files: Vec<(String, String)>,
-    ) -> Result<(), Error>;
+    /// Create a semantic commit from the currently checked out branch.
```
Comment (TomTaehoonKim) [inline repository/src/raw.rs:105]:
```suggestion
    /// Creates a semantic commit from the currently checked out branch.
```

**Comment by TomTaehoonKim** [comment]
No create commands for `Transaction` commits, `AgendaProof` commits, and `ChatLog` commits?

File: repository/src/raw.rs [hunk 2]
```diff
@@ -86,25 +95,20 @@ pub trait RawRepository: Send + Sync {
...
+    async fn create_semantic_commit(&mut self, commit: SemanticCommit)
```
Comment (JeongHunP) [inline repository/src/raw.rs:106]: How commit can be created from semantic_commit? What needs to contain semantic_commit's content in the new commit?

Comment (TomTaehoonKim) [reply repository/src/raw.rs:106]: @JeongHunP Have you checked out `to_semantic_commit` function in this PR? Maybe you're looking for this.

Comment (JeongHunP) [reply repository/src/raw.rs:106]: @TomTaehoonKim I checked but I didn't understand it. So is it right that those title and body in SemanticCommit should be converted to commit message as ASO(simple-git example) shows?

File: repository/src/raw.rs [hunk 3]
```diff
@@ -19,6 +20,14 @@ impl From<git2::Error> for Error {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct SemanticCommit {
```
[the remark this answers is not in the document]
Comment (junha1) [reply repository/src/raw.rs:25]: done.

**Review by TomTaehoonKim — APPROVED** [review]

**Review by JeongHunP — APPROVED** [review]

**Pull request merged**